### ISSUE TITLE
Link UI Incorrectly Anchored when Newly Created

### ISSUE THREAD

**Issue by jeryj** [WordPress/gutenberg#58325]
### Description

Same bug as https://github.com/WordPress/gutenberg/issues/58280, but the fix only works for existing links. When a link is created, and the popover remains open, it is fixed to a point on the screen and doesn't update its position to stay with the link.

### Step-by-step reproduction instructions

1. Add a link on a page with enough content to scroll the editor
2. While the popover is still open from the initial link creation, scroll the page
3. The popover will be fixed in a position on the page instead of staying near the newly created link.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/967608/832c8597-bf45-48e0-b073-62ad1bdcaf19

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by getdave** [comment]
@jeryj @draganescu @scruffian @richtabor I think we ought to roll back the recent Link control PRs. We can bring them back in a single PR which also fixes the bugs. The currently experience in `trunk` is too poor.

**Comment by jeryj** [comment]
I think we should try to fix them up more before reverting. Which bugs are large enough that you want to revert for?

**Comment by getdave** [comment]
I can still replicate the bugs that were uncovered last week in `trunk`. Gutenberg 17.6.0 is released today and the bugs will then be in a release of the Plugin which is not ideal.

Moreover, 9th Feb is Gutenberg 17.7.0 RC which is the final release of the Plugin that will land in WP 6.5. 

I would suggest that unless the fix is _imminent_ then we should:

- revert the changes that caused the bugs
- raising a new PR which _reintroduces_ the changes and includes any necessary bug fixes
- only ship that PR if you can fix the bugs by [Gutenberg 17.7.0 RC on 9th February](https://make.wordpress.org/core/2024/01/19/editor-important-milestones-for-wordpress-6-5/)

What do you think?

**Comment by draganescu** [comment]
@getdave  we should revert:
- [Implementing new UX for invoking rich text Link UI](https://github.com/WordPress/gutenberg/pull/57986/files#top)
- [Fix Link UI popover anchor in rich text](https://github.com/WordPress/gutenberg/pull/58282)

Anything else?

**Comment by jeryj** [comment]
> I can still replicate the bugs that were uncovered last week in trunk

Before we can decide on a revert, I think we need to decide on:
- Which specific bugs are considered big enough to revert
- What the root cause of these bugs are (is it fixed by reverting? Why?)
- Is the overall state of the Link UI better or worse than it was previously

For example, I believe https://github.com/WordPress/gutenberg/issues/58465 is caused by https://github.com/WordPress/gutenberg/pull/47525 and was an issue before the Link UI PRs.

https://github.com/WordPress/gutenberg/issues/58322 is also not a bug caused by the new Link UI, but is more apparent because of it. The root cause is:
- In Firefox
- If focus is within _any_ popover,
- a click happens on the same paragraph that the caret is in
- the caret won't be moved to the new location on the first click

The same bug is there after reverting, but isn't as apparent because previously the link UI could be opened without focus within the popover. Now, we force focus in the popover if it's open, _which is a better experience than before_ as the focus management is predictable.

Are there other bugs that have come up that we're considering reverting for?

**Comment by jeryj** [comment]
Closed by https://github.com/WordPress/gutenberg/pull/58900
